support/cas-server-support-pm/src/main/java/org/apereo/cas/pm/InvalidPasswordException.java

```diff
@@ -0,0 +1,14 @@
+package org.apereo.cas.pm;
+
+/**
+ * Raised by {@link PasswordManagementService} if it is also responsible for validating
+ * passwords and a new password fails validation.
+ *
+ * @author Marcus Watkins
+ * @since 5.2.0
+ */
+
+public class InvalidPasswordException extends RuntimeException {
+
+    private static final long serialVersionUID = 458954862481278L;
+}

```

Comment:
It might be nice to allow for a reason (string/enum/something).  Typically its not displayed to the end user (information leakage), but can be useful in the logs...

Reply:
In general I'd agree, but in this case the consumer of this exception (`PasswordChangeAction`) doesn't know/care the reasons behind it, it's really just a glorified return value. For logging purposes the actually implementer would be responsible.